src/TextRasterizer_P.cpp

```diff
@@ -428,9 +433,71 @@ sk_sp<SkImage> OsmAnd::TextRasterizer_P::rasterize(
     return target.asImage();
 }
 
+void OsmAnd::TextRasterizer_P::drawText(SkCanvas& canvas,
+                                        const TextPaint& textPaint,
+                                        const SkFont& font,
+                                        const SkPaint& paint) const
+{
+#ifndef OSMAND_USE_HARFBUZZ
+    canvas.drawSimpleText(
+        textPaint.text.constData(), textPaint.text.length()*sizeof(QChar), SkTextEncoding::kUTF16,
+        textPaint.positionedBounds.left(), textPaint.positionedBounds.top(),
+        font, paint);
+#else
+    QString inpStr = textPaint.text.toString();
+    auto textQStr = ICU::convertToVisualOrder(inpStr);
+
+    auto hb_font = textPaint.faceData->hbFont.get();
+    hb_font_set_scale(hb_font,
+                      HARFBUZZ_FONT_SIZE_SCALE * font.getSize(),
+                      HARFBUZZ_FONT_SIZE_SCALE * font.getSize());
+    hb_ot_font_set_funcs(hb_font);
+
+    /* Create hb-buffer and populate. */
+    hb_buffer_t* hb_buffer = hb_buffer_create();
+    hb_buffer_add_utf16(hb_buffer, textQStr.utf16(), textQStr.size(), 0, -1);
+    hb_buffer_set_direction(hb_buffer, HB_DIRECTION_LTR);
+    hb_buffer_guess_segment_properties(hb_buffer);
+
+    hb_shape(hb_font, hb_buffer, NULL, 0);
+
+    //  draw glyphs
+    unsigned int length = hb_buffer_get_length(hb_buffer);
+    if (length == 0)
+    {
+        return;
```

Comment:
Maybe hb_buffer needs to be released first?

Reply:
Agree, better to wrap by smart ptr for instance.